### PR TITLE
ci: Add tokens PR action

### PR DESCRIPTION
**Pull request by witwash**
#### Summary

Added an action that would trigger on push in `design-tokens-**` branch. This action will create a PR to `master`.

This action will ensure we have the latest version of tokens from [design-tokens](https://github.com/MewsSystems/design-tokens). Right now we have an action that is committing new files to `design-tokens-${version}` branch. This action will create a PR and generate files that were touched by the tokens update. 

#### Testing steps

We'll need to check whether the trigger is working correctly and the PR is created.

#### Follow-up issues

~~Set up an action that would regenerate `theme_tailor` files in this same PR.~~ done

#### Check during review

- Verify against Jira issue.
- Is the PR over 300 additions? Consider rejecting it with advice to split it. Is it over 500 additions? It should definitely be rejected.
- Unused code removed.
- Build passing.
- Is it a bug fix? Check that it is covered by a proper test (unit or integration).

### CODE REVIEW

**Comment by witwash** [comment]
@ookami-kb, WDYT? 🤔

File: .github/workflows/tokens-pr.yml [hunk 1]
```diff
@@ -0,0 +1,32 @@
...
+        run: melos gen_theme
+
+      - name: Commit changes
+        uses: stefanzweifel/git-auto-commit-action@v5
+        with:
+          commit_message: Automated commit of generated tailor files
+          file_pattern: "*tailor.dart"
+
+      - name: Create Pull Request
+        uses: repo-sync/pull-request@v2
```
Comment (ookami-kb) [inline .github/workflows/tokens-pr.yml:27]: Let's rather use GitHub CLI, it's pre-installed on action runners: https://cli.github.com/manual/gh_pr_create

File: .github/workflows/tokens-pr.yml [hunk 2]
```diff
@@ -0,0 +1,32 @@
...
+        uses: actions/checkout@v2
+
+      - name: Set up Flutter
+        uses: "./.github/actions/setup"
+
+      - name: Generate tokens files
+        run: melos gen_theme
+
+      - name: Commit changes
+        uses: stefanzweifel/git-auto-commit-action@v5
```
Comment (ookami-kb) [inline .github/workflows/tokens-pr.yml:21]: Why do we need a separate action for that? Isn't it just a couple lines of code?

**Review by ookami-kb — CHANGES_REQUESTED** [review]

File: .github/workflows/tokens-pr.yml (hunk 2 above)
Comment (witwash) [reply .github/workflows/tokens-pr.yml:21]: For no particular reason tbh. I wrote it with commands at first, then I looked at the action and it felt cleaner

Comment (ookami-kb) [reply .github/workflows/tokens-pr.yml:21]: I would rather avoid using 3rd party actions unless they bring a lot of benefits.

File: .github/workflows/tokens-pr.yml [hunk 3]
```diff
@@ -0,0 +1,36 @@
...
+
+      - name: Set up Flutter
+        uses: "./.github/actions/setup"
+
+      - name: Generate tokens files
+        run: melos gen_theme
+
+      - name: Commit changes
+        run: |
+          git config --local user.email "tech.account@mewssystems.com"
```
Comment (witwash) [inline .github/workflows/tokens-pr.yml:24]: Do we have a user that we should use here? In `design-tokens` repo we use `tech.account@mewssystems.com`

**Pull request merged**